dhcpcd-hooks/65-kano-dhcp-hook

```diff
@@ -38,24 +36,60 @@ REBOOT | BOUND)
     # Try uploading the tracking data to our servers
     # Should be quiet on failure
     kano-sync --upload-tracking-data --silent &
+}
 
-    # Sync objects from the content API in the background
-    # TODO: This should be removed once we have the daemon done
+# Sync objects from the content API in the background
+# TODO: This should be removed once we have the daemon done
+kano_daemon () {
     kano-content sync &
-
     kano-tracker-ctl +1 'internet-connection-established' &
-    ;;
+}
 
-NOCARRIER | STOPPED)
-    # Are we still networked due to multihomed connection?
+# Check for a multihomed connection and keep it if there
+# otherwise disconnect with a logger notification
+is_multihomed () {
     is_internet
     if [ $? -eq 0 ]; then
-    	logger "kano-dhcp-hook: NOCARRIER/STOPPED but still connected, aborting action"
+    	logger "kano-dhcp-hook: $reason but still connected, aborting action."
     	exit 0
     else
-        logger "kano-dhcp-hook: NOCARRIER/STOPPED and connectivity lost"
+        logger "kano-dhcp-hook: $reason and connectivity lost."
         rm $monitor_file
     fi
+}
+
+# In case of lease RENEWal or REBOUND to a new DHCP server
+# check if connection is sucesfully active and modifly
+# the monitor_file accordingly
+is_reconnected () {
+    if [$? -eq 0]; then
+      logger "kano-dhcp-hook: $reason but still connected...Aborting."
+    exit 0
+    elif ([ $reason = "INFORM" ] && [ $if_up ]); then
+        kano_network_init
+        kano_daemon
+    else
+        logger "kano-dhcp-hook: $ifname down after $reason."
+        rm $monitor_file
+    fi
+}
+
+case "$reason" in
+REBOOT | BOUND)
+    ip_addr=`ip addr show wlan0 | grep inet | awk '{print $2}' | sed -s 's/\/.*//')`
+    logger "kano-dhcp-hook: REBOOT event ipaddr=$ip_addr on iface=$interface"
+    is_connected
+    logger "kano-dhcp-hook: launching network up scripts"
+    kano_network_init
+    kano_daemon
+    ;;
+
+INFORM|RENEW|REBIND|RELEASE)
+    is_internet
+    is_reconnected
     ;;
 
+NOCARRIE|STOP|STOPPED)
```

Comment:
Should this also include `TIMEOUT`, `EXPIRE`, `DEPARTED`, `FAIL`?

Reply:
Probably.

Reply:
On `DEPARTED` and `FAIL` the `dhcpcd` can't run anyway, as it means that there is no interface; therefore no need to be triggered (except if want to be log that through the `logger` - is that a wished feature?). 

The important and undescribed behaviour that was causing the bug, was while a `RENEW` or a `REBOUND` wasn't re-triggering the `dhcpcd` and `wpa_supplicant` as an effect.

`TIMEOUT` keeps already an old lease and in case of `EXPIRED`, it is also handled by default from the `dhclient`, so no need to re-define the default behaviour.